### DISCUSSION TITLE
explain ConfigurationOptions

### DISCUSSION THREAD

**Discussion by yikengboy** [GumTreeDiff/gumtree#366] [Q&A] [answered]
Can you explain what impact adjusting the bu_minsim threshold will have on the output of the editing script?
I debug source code, I found the place where the threshold is used. It seems that the larger the value, the less mapping there is in the mapping.
![image](https://github.com/user-attachments/assets/8323ff62-b59f-4d28-a3ca-ba7e93427275)
I tested it and found that the larger the threshold, the more types the editing script returns. Why is this?

**Comment by jrfaller** [comment, marked as answer]
Hi! I advise you to look into our papers to better understand these thresholds.

Note that there is a new undocumented auto generator that setup this threshold for you automatically at the expense of runtime!

Jean-Rémy Falleri, [Floréal Morandat](https://dblp.org/pid/11/7473.html), [Xavier Blanc](https://dblp.org/pid/37/3159-1.html), [Matias Martinez](https://dblp.org/pid/135/6372.html), [Martin Monperrus](https://dblp.org/pid/62/5292.html):
Fine-grained and accurate source code differencing. [ASE 2014](https://dblp.org/db/conf/kbse/ase2014.html#FalleriMBMM14): 313-324

[Matias Martinez](https://dblp.org/pid/135/6372.html), Jean-Rémy Falleri, [Martin Monperrus](https://dblp.org/pid/62/5292.html):
Hyperparameter Optimization for AST Differencing. [IEEE Trans. Software Eng. 49(10)](https://dblp.org/db/journals/tse/tse49.html#MartinezFM23): 4814-4828 (2023)

Jean-Rémy Falleri, [Matias Martinez](https://dblp.org/pid/135/6372.html):
Fine-grained, accurate and scalable source differencing. [ICSE 2024](https://dblp.org/db/conf/icse/icse2024.html#FalleriM24): 231:1-231:12

**Reply by yikengboy** [reply]
OK, I will study your paper. by the way, how to use this new undocumented auto generator.

**Reply by yikengboy** [reply]
I browsed the implementation of Match and found `SimpleGumtreeAutoMt` matcher,
this matcher using different thresholds, the final output returns the smallest capacity edit script, which is exactly what I want.
Is this what automatic generator you replied to above?

**Reply by jrfaller** [reply]
Yes it's exactly that!

**Reply by yikengboy** [reply]
okay. Thanks a lot. 😁
However, I feel that these thresholds should be dynamically generated first, and then the mapping calculated. Maybe this will make the performance faster.
